### PR TITLE
add boost 1.80.0

### PR DESCRIPTION
**Pull request by tnixeu**
* I've followed [this guide](https://hunter.readthedocs.io/en/latest/creating-new/update.html)
  step by step carefully. **Yes**

I used the 7z instead of the tar.bz2 because it almost 20MB smaller and cmake supports it as well.

### CODE REVIEW

**Review by NeroBurner — APPROVED** [review]
17MB less for this release

If 7z is supported by CMake without a hitch the nearly 20MB less are very welcome

```sh
$ du -h boost_1_80_0.*
94M     boost_1_80_0.7z
111M    boost_1_80_0.tar.bz2
```

**Comment by NeroBurner** [comment]
Thanks for the update and space improvements

**Comment by NeroBurner** [comment]
released: https://github.com/cpp-pm/hunter/releases/tag/v0.24.8

**Comment by hjmallon** [comment]
Unfortunately extracting 7z appears to fail on our CI machines (linux x86_64, cmake 3.24.3). I haven't investigated why yet.

```
CMake Error: Problem with archive_read_next_header(): Pathname cannot be converted from UTF-16LE to current locale.
CMake Error: Problem extracting tar: /root/.hunter/_Base/Download/Boost/1.80.0/5463e53/boost_1_80_0.7z
```

**Comment by NeroBurner** [comment]
TL;DR: set the `LANG="C.UTF-8"` env-variable in the Dockerfile (or any other way)

```
# set LANG to fix extracting of 7z Boost 1.80 archive by CMake during Hunter build
ENV LANG "C.UTF-8"
```

---

Long version:
We had the same issue in our docker images.

All hunter Boost builds with Boost version 1.80.0 fail because this version switched to use the 7z archive to save space.

This works well on cpp-pm/hunter and locally when running inside my Desktop machine. But when running the same command inside our docker container the following error shows:

```
-- extracting... [tar xfz]
CMake Error: Problem with archive_read_next_header(): Pathname cannot be converted from UTF-16LE to current locale.
CMake Error: Problem extracting tar: /builds/hunter/hunter/_Base/Download/Boost/1.80.0/5463e53/boost_1_80_0.7z
```

The found workaround/fix is to set the `LANG` environment variable to `C.UTF-8`

The current locale command inside the docker container returns the following:

```sh
$ locale
LANG=
LANGUAGE=
LC_CTYPE="POSIX"
LC_NUMERIC="POSIX"
LC_TIME="POSIX"
LC_COLLATE="POSIX"
LC_MONETARY="POSIX"
LC_MESSAGES="POSIX"
LC_PAPER="POSIX"
LC_NAME="POSIX"
LC_ADDRESS="POSIX"
LC_TELEPHONE="POSIX"
LC_MEASUREMENT="POSIX"
LC_IDENTIFICATION="POSIX"
LC_ALL=
```

When setting the `LANG` variable it looks as follows (and then CMake extract works again)

```sh
$ export LANG="C.UTF-8"
$ locale
LANG=C.UTF-8
LANGUAGE=
LC_CTYPE="C.UTF-8"
LC_NUMERIC="C.UTF-8"
LC_TIME="C.UTF-8"
LC_COLLATE="C.UTF-8"
LC_MONETARY="C.UTF-8"
LC_MESSAGES="C.UTF-8"
LC_PAPER="C.UTF-8"
LC_NAME="C.UTF-8"
LC_ADDRESS="C.UTF-8"
LC_TELEPHONE="C.UTF-8"
LC_MEASUREMENT="C.UTF-8"
LC_IDENTIFICATION="C.UTF-8"
LC_ALL=
```

For a permanent fix the following lines are added to the `Dockerfile`:

```
# set LANG to fix extracting of 7z Boost 1.80 archive by CMake during Hunter build
ENV LANG "C.UTF-8"
```
